src/toast/ops/simple_statcut.py

```diff
@@ -0,0 +1,276 @@
+from ..utils import Environment, Logger, flagged_noise_fill, name_UID
+from .operator import Operator
+
+
+@trait_docs
+class SimpleStatCut(Operator):
+    """An operator that flags extreme detector samples."""
+
+    # Class traits
+
+    API = Int(0, help="Internal interface version for this operator")
+
+    det_data = Unicode(defaults.det_data, help="Observation detdata key to analyze")
+
+    det_mask = Int(
+        defaults.det_mask_invalid,
+        help="Bit mask value for per-detector flagging",
+    )
+
+    det_flags = Unicode(
+        defaults.det_flags,
+        allow_none=True,
+        help="Observation detdata key for flags to use",
+    )
+
+    det_flag_mask = Int(
+        defaults.det_mask_invalid,
+        help="Bit mask value for detector sample flagging",
+    )
+
+    shared_flags = Unicode(
+        defaults.shared_flags,
+        allow_none=True,
+        help="Observation shared key for telescope flags to use",
+    )
+
+    shared_flag_mask = Int(
+        defaults.shared_mask_nonscience,
+        help="Bit mask value for optional shared flagging",
+    )
+
+    view = Unicode(
+        None,
+        allow_none=True,
+        help="Find glitches in this view",
+    )
+
+    medfilt_kernel_size = Int(
+        101,
+        help="Median filter kernel width.  Either 0 (full interval) "
+        "or a positive odd number",
+    )
+
+    limit = Float(
+        3.0,
+        help="Distance from median to be considered pathological",
+    )
+
+    out = Unicode(
+        "stats",
+        help="Observation key for statistics",
+    )
+
+    @traitlets.validate("det_mask")
+    def _check_det_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Det mask should be a positive integer")
+        return check
+
+    @traitlets.validate("shared_flag_mask")
+    def _check_shared_flag_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Shared flag mask should be a positive integer")
+        return check
+
+    @traitlets.validate("det_flag_mask")
+    def _check_det_flag_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Det flag mask should be a positive integer")
+        return check
+
+    @traitlets.validate("medfilt_kernel_size")
+    def _check_medfilt_kernel_size(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("medfilt_kernel_size cannot be negative")
+        if check > 0 and check % 2 == 0:
+            raise traitlets.TraitError("medfilt_kernel_size cannot be even")
+        return check
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.net_factors = []
+        self.total_factors = []
+        self.weights_in = []
+        self.weights_out = []
+        self.rates = []
```

Comment:
It looks like these lines are left over from a copy / paste?

Reply:
Thanks!